docs/spec_draft.md

```diff
@@ -3406,6 +3407,117 @@ More formally, `results[:][result_index] = reduce(input_slices)` where:
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.reduce_scatter
+
+### Semantics
+
+<img align="center" src="spec_draft/reduce_scatter.svg" />
+
+Within each process group in the StableHLO grid, performs reduction, using
+`computations`, over the values of the `operand` tensor from each process,
+splits the reduction result along `scatter_dimension` into parts, and scatters
+the split parts between the processes to produce the `result`.
+
+The operation splits the StableHLO grid into `process_groups` as follows:
+  * `channel_id <= 0` and `use_global_device_ids = false`,
+    `cross_replica(replica_groups)`.
+  * `channel_id > 0` and `use_global_device_ids = false`,
+    `cross_replica_and_partition(replica_groups)`.
+  * `channel_id > 0` and `use_global_device_ids = true`,
+    `flattened_ids(replica_groups)`.
+
+Within each `process_group`:
+  * ```
```

Comment:
can we make this one pythonic pseudo code.

Reply:
The three sections are in sync with the three parts of the semantics. Let us seek for additional review around it.

Reply:
consider putting all the separate bullet points into one section and separate them by line breaks or comments

```
# 1.
reduced_value ...

# 2. 
split_count ...

# 3.
receiver ...
```